chia/wallet/wallet_state_manager.py

```diff
@@ -529,13 +528,37 @@ async def synced(self):
             return True
         return False
 
-    def set_sync_mode(self, mode: bool, sync_height: uint32 = uint32(0)):
-        """
-        Sets the sync mode. This changes the behavior of the wallet node.
-        """
-        self.sync_mode = mode
-        self.sync_target = sync_height
-        self.state_changed("sync_changed")
+    @property
+    def sync_mode(self) -> bool:
+        return self._sync_target is not None
+
+    @property
+    def sync_target(self) -> Optional[uint32]:
+        return self._sync_target
+
+    @asynccontextmanager
+    async def set_sync_mode(self, target_height: uint32) -> AsyncIterator[None]:
+        self.log.debug(f"set_sync_mode enter {await self.blockchain.get_finished_sync_up_to()}-{target_height}")
```

Comment:
you end up calling this `async` function here unconditionally (for the 3rd time), regardless of whether we have debug logging enabled or not. I think you should try to avoid these redundant calls

Reply:
Its not that this gets called frequently but i wrapped them with log level checks in [edcb194](https://github.com/Chia-Network/chia-blockchain/pull/14524/commits/edcb1940dd8c41b5bedccaf9523b4f3fe3d23336). I have a branch where i work on some improvements to this status, like keeping track of how many times we synced up to the latest height, to allow dropping the secondary sync task which also includes making this cached values, then we can drop the log level checks again.